src/cryptography/hazmat/primitives/serialization/pkcs7.py

```diff
@@ -263,6 +263,116 @@ def encrypt(
         return rust_pkcs7.encrypt_and_serialize(self, encoding, options)
 
 
+class PKCS7EnvelopeDecryptor:
```

Comment:
Hmm, I wonder if we need a builder for this case, is `pkcs7_decrypt(data, recipient_cert, recipient_key, options)` sufficient, or might we want other things?

cc: @reaperhulk

Reply:
TBH on my side I was more focused on developing code that worked, so any API design is fine to me. I'd probably go for the unique function as it's the simplest one.

Reply:
Two ideas I'm thinking about right now:
- Should we allow multiple certs / pkeys as arguments and let `pkcs7_decrypt` choose the (or a) certificate that works? Outlook does that for instance (I've got multiple certificates associated with different mailboxes).
- In your case, you're missing the `encoding` parameter (DER, PEM, SMIME). Are we expecting `pkcs7_decrypt` to figure it out by itself?

Reply:
I'd bias to the function as well. I would not want the function to guess though -- we should either have a param or 3 separate functions.

Reply:
Ah yes, everywhere else in our API we have seperate functions for DER vs. PEM.

Reply:
Just pushed first implementation of `pkcs7_decrypt_{encoding}` 😄